lib_static/open_project/authentication/strategies/warden/doorkeeper_oauth.rb

```diff
@@ -42,7 +42,7 @@ def access_token_invalid?(access_token)
           end
 
           def authenticate_user(id)
-            user = id && User.active.find_by(id:)
+            user = id && User.active.where(id:).or(ServiceAccount.where(id:)).first
```

Comment:
:yellow_circle: I think that should not be necessary after our fixes to lazy loading anymore, right?

A service account **is** a user and thus `User.active.where(:id)` should find it.

As a side note, if I read this code here correctly, it would not filter service accounts by the `active` scope, which would be a bug as far as I can tell.

Reply:
For some reason I have an extra hard time finding the implementation of `User.active`, but obviously this is surprising behaviour. I would not have expected that it makes a difference whether I do `User.active` or `User.all.select(&:active?)`, except for the former to be more efficient.

Meh, that's a bummer...